.wowhead2/wowhead_fetcher.py

```
@@ -0,0 +1,520 @@
+    print(f"Fetching {total_to_fetch} entities")
+
+    try:
+      with ThreadPoolExecutor(max_workers=self.max_workers) as executor:
+        # Submit all fetch tasks to the thread pool
+        future_to_entity = {executor.submit(_fetch_worker, entity, self.locale, self._stop_event): entity for entity in entities_to_fetch}
+
+        for i, future in enumerate(as_completed(future_to_entity), 1):
+          # Check if we should stop before processing results
+          if self._stop_event.is_set():
+            print(f"Stop requested - cancelling remaining {total_to_fetch - i + 1} tasks")
+            print(f"This might take a minute or two, please wait not to corrupt the database...")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)